### PR TITLE
copy feature available using react hot toast and navigator.clipboard

### LINKED ISSUE #39: Add list sharing feature
The user should be able to share with other people. 

To do this we need to extract the token id or/and show it on the screen or add a button to copy the id to the clipboard.

### PR DESCRIPTION
**Pull request by mira-kine**
## Description

This PR has a button that allows a user to copy the listToken (list name) onto clipboard. It uses react hot toast to notify the user that a name was copied successfully, and the button changes from "Copy List Name" to "Copied!" for 2 seconds. I am open to changing it for better accessibility, but for now the functionality of copying it onto clipboard should be working. I also assumed that by "sharing list feature" we just wanted a way to copy it onto the clipboard easily instead of highlighting + c + v. We can style it better in the `List` styling issue. 

## Related Issue

Closes #39 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

1. From your local machine git checkout main then git pull
2. git checkout MK-copy-list-feature
3. npm run start
4. Go to `List` and click on "Copy List Name"
5. A pop up should come up saying "Copied successfully" and the button should change from "Copy List Name" to "Copied!"
6. Go back to Home after logging out, and try out the newly copied list name that is on your clipboard. You can also just paste it somewhere to see that it was copied onto your machine.

### CODE REVIEW

File: src/views/List.jsx [hunk 1]
```diff
@@ -68,15 +71,26 @@ export function List({ data, listToken, loading, logOut }) {
...
+		navigator.clipboard.writeText(`${listToken}`);
+		toast.success('Copied!');
+		setTimeout(() => {
+			setCopy(false);
+		}, 2000);
+	}
 
 	return (
 		<div className="list-container">
+			<Toaster />
```
Comment (Trevor-Rezac) [inline src/views/List.jsx:85]: I think you might be able to wrap Toaster around App in Index.jsx and then toast will be available on every component of the app.

**Review by N-Zubko — APPROVED** [review]
For me 2 seconds is enough to see the notification, plus the button does the main purpose - copies the list token into the clipboard, which is more important. So I am happy to approve the PR!

**Comment by N-Zubko** [comment]
You could mention that the toast needs to be installed first before testing the app

**Review by Trevor-Rezac — APPROVED** [review]
Approving this! My only comments were just a suggestion about Toaster.

**Review by lindseyindev — APPROVED** [review]
I like that you added this functionality to the app and nice work on installing React Hot Toast and building out the copy-to-clipboard functionality!  

I was trying to find a way to handle the conditional render other than setTimeout, I've just found setTimeout to be the cause of a lot of memory leaks (which can slow the performance of the app) and buggy behavior when used excessively and when the timeouts aren't cleared on component unmount.

I couldn't find an alternate solution to this particular functionality without using setTimeout and I also increased the timeout to a really large number and navigated away from the page and didn't see any memory leak warnings so I am happy with that -- just something to be on the lookout for in the future.

File: src/views/List.jsx (hunk 1 above)
Comment (mira-kine) [reply src/views/List.jsx:85]: Good call!

**Pull request merged**